src/Adapter/Presenter/Module/PaymentModulesPresenter.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Presenter\Module;
+
+use PrestaShop\PrestaShop\Adapter\Module\PaymentModuleListProvider;
+use PrestaShop\PrestaShop\Adapter\Presenter\PresenterInterface;
+
+/**
+ * Class PaymentModulesPresenter is responsible for presenting payment modules.
+ */
+class PaymentModulesPresenter
+{
+    /**
+     * @var PresenterInterface
+     */
+    private $modulePresenter;
+
+    /**
+     * @var PaymentModuleListProvider
+     */
+    private $paymentModuleListProvider;
+
+    /**
+     * @param PresenterInterface $modulePresenter
+     * @param PaymentModuleListProvider $paymentModuleListProvider
+     */
+    public function __construct(
+        PresenterInterface $modulePresenter,
+        PaymentModuleListProvider $paymentModuleListProvider
+    ) {
+        $this->modulePresenter = $modulePresenter;
+        $this->paymentModuleListProvider = $paymentModuleListProvider;
+    }
+
+    /**
+     * Get presented payment modules.
+     *
+     * @return array
+     */
+    public function present(): array
+    {
+        $installedModules = $this->paymentModuleListProvider->getPaymentModuleList();
```

Comment:
Seems like this provider only relies on Addons repository calls, no problem for this version But I guess we'll need to be careful when we're gonna merge this into the next version because of the split right?

@PierreRambaud wdyt?

Reply:
Yes exactly